.github/workflows/rust.yml

```
@@ -0,0 +1,146 @@
+name: Rust
+
+on:
+  push:
+    branches:
+      - master
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build:
+    name: Build Napture (Ubuntu)
+    runs-on: ubuntu-24.04
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install Dependencies
+      run: |
+        echo "Installing dependencies..."
+        sudo apt-get update
+        sudo apt-get install -y libglib2.0-dev libcairo2-dev libgraphene-1.0-dev libgtk-4-dev libadwaita-1-dev 
+        sudo add-apt-repository universe
+        sudo apt install libfuse2
+        echo "Dependencies installed."
+
+    - name: Build Napture
+      run: |
+        echo "Building Napture..."
+        cd napture
+        export PKG_CONFIG_PATH=/usr/lib/pkgconfig
+        cargo build --release
+        echo "Napture built successfully."
+    - name: Install linuxdeploy
+      run: |
+        echo "Starting artifact upload phase..."
+        echo "Installing linuxdeploy..."
+        wget https://github.com/linuxdeploy/linuxdeploy/releases/download/continuous/linuxdeploy-x86_64.AppImage
+        chmod +x linuxdeploy-x86_64.AppImage
+        echo "linuxdeploy installed."
+    - name: Create Napture AppImage
+      run: |
+        echo "Creating Napture AppImage..."
+        cd napture
+        mv target/release/webx target/release/napture
+        ARCH=x86_64
+         ../linuxdeploy-x86_64.AppImage --appdir AppDir -e target/release/napture -d io.github.face_hh.Napture.desktop -i io.github.face_hh.Napture.svg
```

Crap, I forgot to use the `appimage` plugin so it outputs an executable
```suggestion
         ../linuxdeploy-x86_64.AppImage --appdir AppDir -e target/release/napture -d io.github.face_hh.Napture.desktop -i io.github.face_hh.Napture.svg -o appimage
```